Automattic/jetpack
# [memberships] Remove token from url

Fixes https://github.com/Automattic/gold/issues/169

## Proposed changes:

* Remove adding the token to the URL for no reason
* Do not rely on `subscribe=success` to know the subscription status of the current user
* Small refactor

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







WPCOM
* Push this branch on your sandbox and sandbox bestmemberships.wordpress.com
* Open https://bestmemberships.wordpress.com/2023/08/31/access-only-for-free-subscribers/
* From incognito, subscribe for free with a new user
* Do not confirm the email
* Make sure the token is not in the URL
* Make sure you see this:
![Uploading Screenshot 2023-10-11 at 18.23.53.png…]()
* Reload the page, you should see the same thing, as the cookie should be 
* Close the window 
* Open a new incognito and re-enter the email, you should see the same
* Answer the confirmation email, you now should have access


Self-hosted
* Create a new JN instance
* Connect Earn to Stripe
* Create a post with Newsletter access and one with a paid-content block
* Pay for the access and when clicking "Continue" make sure the URL does not contain the token
